instagram/src/components/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Comment from "./Comment";
+import CommentInput from "./CommentInput";
+let count = 0;
+class CommentSection extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      comments: props.comments,
+      comment: "",
+      count: count,
+      thisUserClicked: false
+    };
+  }
+  componentDidMount() {
+    const id = this.props.postId;
+    // alert("Got Item!");
+    if (localStorage.getItem(id)) {
+      this.setState({
+        comment: JSON.parse(localStorage.getItem(this.props.postId))
+      }); //might need to tweek
+    }
+  }
+
+  componentWillMount() {
+    this.setComments();
+  }
+
+  setComments = () => {
+    localStorage.setItem(
+      this.props.postId,
+      JSON.stringify(this.state.comments)
+    );
+  };
+
+  commentSubmitHandle = event => {
+    event.preventDefault();
+    const newComment = {
+      username: "ljhig",
+      text: this.state.comment
+    };
+    const comments = this.state.comments.slice();
+    comments.push(newComment);
+    this.setState({ comments, text: "" });
```

Comment:
there is no this.state.text. there is a newComment.text, but that is locally scoped to this method.